src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -501,148 +1387,699 @@ private boolean isAmqpHeaderValid(
+                        receiverSettleMode);
+                }
+            }
+
+            private void doApplicationBegin(
+                long traceId,
+                long authorization,
+                long affinity,
+                String address,
+                AmqpRole role,
+                AmqpSenderSettleMode senderSettleMode,
+                AmqpReceiverSettleMode receiverSettleMode)
+            {
+                assert state == 0;
+                state = AmqpState.openingInitial(state);
+
+                router.setThrottle(initialId, this::onApplicationInitial);
+
+                final AmqpBeginExFW beginEx = amqpBeginExRW.wrap(extBuffer, 0, extBuffer.capacity())
+                    .typeId(amqpTypeId)
+                    .containerId(containerId)
+                    .channel(channelId)
+                    .address(address)
+                    .role(r -> r.set(amqpRole(role)))
+                    .senderSettleMode(s -> s.set(amqpSenderSettleMode(senderSettleMode)))
+                    .receiverSettleMode(r -> r.set(amqpReceiverSettleMode(receiverSettleMode)))
+                    .build();
+
+                doBegin(application, newRouteId, initialId, traceId, authorization, affinity, beginEx);
+
+                if (initialSlot == NO_SLOT)
+                {
+                    initialSlot = bufferPool.acquire(initialId);
+                }
+
+                if (initialSlot == NO_SLOT)
+                {
+                    cleanup(traceId, authorization);
+                }
+            }
+
+            private void doApplicationData(
+                long traceId,
+                long authorization,
+                AmqpRole role)
+            {
+                assert AmqpState.initialOpening(state);
+
+                switch (role)
+                {
+                case SENDER:
+                    // TODO
+                    break;
+                }
+            }
+
+            private void doApplicationAbort(
+                long traceId,
+                long authorization,
+                Flyweight extension)
+            {
+                setInitialClosed();
+
+                doAbort(application, newRouteId, initialId, traceId, authorization, extension);
+            }
+
+            private void doApplicationAbortIfNecessary(
+                long traceId,
+                long authorization)
+            {
+                if (!AmqpState.initialClosed(state))
+                {
+                    doApplicationAbort(traceId, authorization, EMPTY_OCTETS);
+                }
+            }
+
+            private void setInitialClosed()
+            {
+                assert !AmqpState.initialClosed(state);
+
+                state = AmqpState.closeInitial(state);
+                cleanupInitialSlotIfNecessary();
+
+                if (AmqpState.closed(state))
+                {
+                    // TODO
+                }
+            }
+
+            private void onApplicationInitial(
```

Comment:
Rename to `onApplication`, as it now covers both initial and reply application frames.